### PR TITLE
Added Import Assessment Json

### PR DESCRIPTION
**Pull request by aperfilev**
Adding import migration button with capability to open stored assessment in *.json format.

Current functionality for scanning json file and jump forward to New Migration Step 2. Allowing to observe migration details.

[Task 2369198](https://msdata.visualstudio.com/Database%20Systems/_workitems/edit/2369198): [DMA Exit] | Re/Import JSON features UI design
Screenshot 2023-06-29 124856-1
Screenshot 2023-07-31 095807

### CODE REVIEW

File: extensions/sql-migration/src/dashboard/tabBase.ts [hunk 1]
```diff
@@ -136,6 +137,33 @@ export abstract class TabBase<T> implements azdata.Tab, vscode.Disposable {
...
+				label: loc.DESKTOP_IMPORT_MIGRATION_BUTTON_LABEL,
+				description: loc.DESKTOP_IMPORT_MIGRATION_BUTTON_DESCRIPTION,
+				height: 24,
+				iconHeight: 24,
+				iconWidth: 24,
+				iconPath: IconPathHelper.addNew,
+			}).component();
+		this.disposables.push(
+			importMigrationButton.onDidClick(async () => {
+				const file = await utils.promptUserForFile({ 'Json (*.json)': ['json'] });
```
Comment (brian-harris) [inline extensions/sql-migration/src/dashboard/tabBase.ts:153]: 'Json ...' should be added as a resource string.

File: extensions/sql-migration/src/dashboard/tabBase.ts [hunk 2]
```diff
@@ -136,6 +137,33 @@ export abstract class TabBase<T> implements azdata.Tab, vscode.Disposable {
...
+				if (file) {
+					void vscode.window.showInformationMessage("Selected import file: " + file);
```
Comment (brian-harris) [inline extensions/sql-migration/src/dashboard/tabBase.ts:155]: same here, user message should be added as a resource string.

File: .gitignore [hunk 3]
```diff
@@ -1,5 +1,6 @@
 .DS_Store
 .cache
+.idea
```
Comment (brian-harris) [inline .gitignore:3]: is this change needed

Comment (aperfilev) [reply .gitignore:3]: this removes project files

Comment (raymondtruong) [reply .gitignore:3]: I would run this one by the ADS team, as this repo is shared by the entire product so this may have unintended consequences.

+ @Charles-Gagnon any thoughts?

File: extensions/sql-migration/src/dashboard/dashboardTab.ts [hunk 4]
```diff
@@ -143,6 +143,7 @@ export class DashboardTab extends TabBase<DashboardTab> {
 		toolbar.addToolbarItems([
 			<azdata.ToolbarComponent>{ component: this.createNewMigrationButton() },
 			<azdata.ToolbarComponent>{ component: this.createNewLoginMigrationButton() },
+			<azdata.ToolbarComponent>{ component: this.createImportMigrationButton() },
```
Comment (raymondtruong) [inline extensions/sql-migration/src/dashboard/dashboardTab.ts:146]: if this is the first PR of several for this feature, should we disable / comment out the button for now until it's ready? Or did you plan on iterating on this PR and not checking in until it's ready?

File: extensions/sql-migration/src/constants/strings.ts [hunk 5]
```diff
@@ -1427,6 +1427,8 @@ export const MIGRATION_SERVICE_DESCRIPTION = localize('sql.migration.select.serv
 
 // Desktop tabs
 export const DESKTOP_MIGRATION_BUTTON_LABEL = localize('sql.migration.tab.button.migration.label', 'New migration');
+export const DESKTOP_IMPORT_MIGRATION_BUTTON_LABEL = localize('sql.migration.tab.button.migration.label', 'Import migration');
```
Comment (raymondtruong) [inline extensions/sql-migration/src/constants/strings.ts:1430]: the "ID" (first parameter to the localize function) needs to be unique for every string

File: .gitignore (hunk 3 above)
Comment (Charles-Gagnon) [reply .gitignore:3]: I don't think anyone uses IntelliJ and not sure why anyone would for ADS. That being said I don't really have a problem with adding it if someone is actually using it.

@kisantia @kburtram FYI

Comment (kisantia) [reply .gitignore:3]: is this change actually necessary for this PR? If it isn't related to the other changes in this PR, please send a separate PR with an explanation for changing something like this.

Comment (aperfilev) [reply .gitignore]: These unrelated changes rolled back

File: extensions/sql-migration/src/dashboard/dashboardTab.ts (hunk 4 above)
Comment (aperfilev) [reply extensions/sql-migration/src/dashboard/dashboardTab.ts:146]: Correct, I plan to iterate on this PR and check in until its ready. Will remove draft when ready then.

**Comment by cheenamalhotra** [comment]
I think your changes are rebased to main, and therefore this PR is re-committing changes which shouldn't be done ideally. It becomes extremely hard to review the changes intended in this PR if something changed in-between or during merge.

Is it possible to open a new PR based off main branch by cherry-picking only the commits made in this PR into a new branch in the order they were committed?

File: extensions/azurecore/src/account-provider/auths/azureAuth.ts [hunk 6]
```diff
@@ -399,87 +234,10 @@ export abstract class AzureAuth implements vscode.Disposable {
...
-		await this.saveTokenAdal(tenant, resource, accountKey, result);
-
-		return result;
-	}
+	//#region tenant calls
 
-	public async getTenantsMsal(token: string, tokenClaims: TokenClaims): Promise<Tenant[]> {
-		const tenantUri = url.resolve(this.metadata.settings.armResource.endpoint, 'tenants?api-version=2020-01-01');
+	public async getTenants(token: string, tokenClaims: TokenClaims): Promise<Tenant[]> {
+		const tenantUri = url.resolve(this.metadata.settings.armResource.endpoint, 'tenants?api-version=2019-11-01');
```
Comment (Ramudaykumar) [inline extensions/azurecore/src/account-provider/auths/azureAuth.ts:240]: why are we going to older API version?

**Comment by stuti149** [comment]
Closing this PR as it contains the commits which are not specific to this feature. I will raise a new PR which has only changes for Import Assessment Json feature.
